### ISSUE TITLE
Support for extensions with version number being 10 or greater

### ISSUE THREAD

**Issue by gavinsawyer** [firebase/firebase-tools#6302]
### Environment info

**firebase-tools:** 12.5.2

**Platform:** macOS

### Test case

Upload a Firebase Extension with a version number being 10 or greater, including `version` in extension.yaml and entry in CHANGELOG.md. Receive an error stating `No entry for version 10.X.X found in CHANGELOG.md.`

### Steps to reproduce

1. Modify any firebase extension for this version number
2. Run `firebase ext:dev:upload ${EXTENSION_PUBLISHER_ID}/${EXTENSION_ID} --ref main`, select any release stage
3. Receive error

### Expected behavior

Find existing entry in CHANGELOG.md

### Actual behavior

Does not seem to match version number as entered in CHANGELOG.md

```
[debug] [2023-08-25T19:25:28.517Z] ----------------------------------------------------------------------
[debug] [2023-08-25T19:25:28.518Z] Command:       /usr/local/bin/node /Users/gavinsawyer/Developer/firebase-web-authn/node_modules/.bin/firebase ext:dev:upload gavinsawyer/firebase-web-authn --ref main --debug
[debug] [2023-08-25T19:25:28.518Z] CLI Version:   12.5.2
[debug] [2023-08-25T19:25:28.518Z] Platform:      darwin
[debug] [2023-08-25T19:25:28.518Z] Node Version:  v18.17.1
[debug] [2023-08-25T19:25:28.519Z] Time:          Fri Aug 25 2023 15:25:28 GMT-0400 (Eastern Daylight Time)
[debug] [2023-08-25T19:25:28.519Z] ----------------------------------------------------------------------
[debug] 
[debug] [2023-08-25T19:25:28.548Z] Object ".extensions" in "firebase.json" has unknown property: {"additionalProperty":"firebase-web-authn"}
[debug] [2023-08-25T19:25:28.550Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2023-08-25T19:25:28.550Z] > authorizing via signed-in user (g@vinsawyer.dev)
[info] i  extensions: ensuring required API firebaseextensionspublisher.googleapis.com is enabled... 
[debug] [2023-08-25T19:25:28.551Z] >>> [apiv2][query] GET https://serviceusage.googleapis.com/v1/projects/fir-web-authn/services/firebaseextensionspublisher.googleapis.com [none]
[debug] [2023-08-25T19:25:28.551Z] >>> [apiv2][(partial)header] GET https://serviceusage.googleapis.com/v1/projects/fir-web-authn/services/firebaseextensionspublisher.googleapis.com x-goog-quota-user=projects/fir-web-authn
[debug] [2023-08-25T19:25:29.146Z] <<< [apiv2][status] GET https://serviceusage.googleapis.com/v1/projects/fir-web-authn/services/firebaseextensionspublisher.googleapis.com 200
[debug] [2023-08-25T19:25:29.147Z] <<< [apiv2][body] GET https://serviceusage.googleapis.com/v1/projects/fir-web-authn/services/firebaseextensionspublisher.googleapis.com [omitted]
[info] ✔  extensions: required API firebaseextensionspublisher.googleapis.com is enabled 
[debug] [2023-08-25T19:25:29.148Z] >>> [apiv2][query] GET https://firebaseextensionspublisher.googleapis.com/v1beta/projects/-/publisherProfile publisherId=gavinsawyer
[debug] [2023-08-25T19:25:29.646Z] <<< [apiv2][status] GET https://firebaseextensionspublisher.googleapis.com/v1beta/projects/-/publisherProfile 200
[debug] [2023-08-25T19:25:29.647Z] <<< [apiv2][body] GET https://firebaseextensionspublisher.googleapis.com/v1beta/projects/-/publisherProfile {"name":"projects/709287606712/publisherProfile","publisherId":"gavinsawyer","registerTime":"2023-05-12T17:31:26.418657Z","displayName":"Gavin Sawyer","websiteUri":"https://gavinsawyer.dev","state":"ACTIVE"}
[debug] [2023-08-25T19:25:29.649Z] >>> [apiv2][query] GET https://firebase.googleapis.com/v1beta1/projects/709287606712 [none]
[debug] [2023-08-25T19:25:30.057Z] <<< [apiv2][status] GET https://firebase.googleapis.com/v1beta1/projects/709287606712 200
[debug] [2023-08-25T19:25:30.057Z] <<< [apiv2][body] GET https://firebase.googleapis.com/v1beta1/projects/709287606712 {"projectId":"gavinsawyer-bf631","projectNumber":"709287606712","displayName":"gavinsawyer-bf631","name":"projects/gavinsawyer-bf631","resources":{"hostingSite":"gavinsawyer-bf631"},"state":"ACTIVE","etag":"1_96eaa66b-14c4-454c-89ba-a0893b8334b9"}
[debug] [2023-08-25T19:25:30.058Z] Checking if latest publisher TOS has been accepted by 709287606712...
[debug] [2023-08-25T19:25:30.058Z] >>> [apiv2][query] GET https://firebaseextensionstos-pa.googleapis.com/v1/projects/709287606712/publishertos [none]
[debug] [2023-08-25T19:25:30.464Z] <<< [apiv2][status] GET https://firebaseextensionstos-pa.googleapis.com/v1/projects/709287606712/publishertos 200
[debug] [2023-08-25T19:25:30.464Z] <<< [apiv2][body] GET https://firebaseextensionstos-pa.googleapis.com/v1/projects/709287606712/publishertos {"name":"projects/709287606712/publishertos","lastAcceptedVersion":"v1","lastAcceptedTime":"2023-05-12T17:32:43.356380Z","latestTosVersion":"v1"}
[debug] [2023-08-25T19:25:30.464Z] Already accepted version v1 of Extensions publisher TOS.
[debug] [2023-08-25T19:25:30.465Z] >>> [apiv2][query] GET https://firebaseextensionspublisher.googleapis.com/v1beta/publishers/gavinsawyer/extensions/firebase-web-authn [none]
[debug] [2023-08-25T19:25:31.697Z] <<< [apiv2][status] GET https://firebaseextensionspublisher.googleapis.com/v1beta/publishers/gavinsawyer/extensions/firebase-web-authn 200
[debug] [2023-08-25T19:25:31.697Z] <<< [apiv2][body] GET https://firebaseextensionspublisher.googleapis.com/v1beta/publishers/gavinsawyer/extensions/firebase-web-authn {"name":"publishers/gavinsawyer/extensions/firebase-web-authn","ref":"gavinsawyer/firebase-web-authn","latestVersion":"9.6.6","latestVersionCreateTime":"2023-08-08T21:16:21.806149Z","state":"PUBLISHED","createTime":"2023-05-12T18:10:54.856431Z","iconUri":"https://storage.googleapis.com/firebase-extensions-icons/extension_icons/gavinsawyer/firebase-web-authn_9.6.6@512.png","publisher":{"displayName":"Gavin Sawyer"},"repoUri":"https://github.com/gavinsawyer/firebase-web-authn","tags":["utilities"],"latestApprovedVersion":"9.6.6"}
[debug] [2023-08-25T19:25:31.699Z] >>> [apiv2][query] GET https://firebaseextensionspublisher.googleapis.com/v1beta/publishers/gavinsawyer/extensions/firebase-web-authn/versions/latest [none]
[debug] [2023-08-25T19:25:32.106Z] <<< [apiv2][status] GET https://firebaseextensionspublisher.googleapis.com/v1beta/publishers/gavinsawyer/extensions/firebase-web-authn/versions/latest 200
[debug] [2023-08-25T19:25:32.107Z] <<< [apiv2][body] GET https://firebaseextensionspublisher.googleapis.com/v1beta/publishers/gavinsawyer/extensions/firebase-web-authn/versions/latest ...
[info] 
Extension: gavinsawyer/firebase-web-authn
State: Published
Latest Version: 9.6.6
Version in Extensions Hub: 9.6.6
Source in GitHub: https://github.com/gavinsawyer/firebase-web-authn/dist/libs/extension (use --repo and --root to modify)

[info] Validating source code at https://github.com/gavinsawyer/firebase-web-authn/tree/main/dist/libs/extension...
[debug] [2023-08-25T19:25:33.367Z] Data is 523632

...

[debug] [2023-08-25T19:25:33.657Z] [unzip] Entry: firebase-web-authn-main/dist/ (compressed_size=0 bytes, uncompressed_size=0 bytes)
[debug] [2023-08-25T19:25:33.657Z] [unzip] Processing entry: firebase-web-authn-main/dist/
[debug] [2023-08-25T19:25:33.657Z] [unzip] mkdir: /var/folders/j2/g1020gyn5z56cn_cxf7z8c1w0000gn/T/tmp-21586-pfgwn8f5lo3M/firebase-web-authn-main/dist/
[debug] [2023-08-25T19:25:33.657Z] [unzip] Entry: firebase-web-authn-main/dist/libs/ (compressed_size=0 bytes, uncompressed_size=0 bytes)
[debug] [2023-08-25T19:25:33.657Z] [unzip] Processing entry: firebase-web-authn-main/dist/libs/
[debug] [2023-08-25T19:25:33.657Z] [unzip] mkdir: /var/folders/j2/g1020gyn5z56cn_cxf7z8c1w0000gn/T/tmp-21586-pfgwn8f5lo3M/firebase-web-authn-main/dist/libs/
[debug] [2023-08-25T19:25:33.657Z] [unzip] Entry: firebase-web-authn-main/dist/libs/extension/ (compressed_size=0 bytes, uncompressed_size=0 bytes)
[debug] [2023-08-25T19:25:33.657Z] [unzip] Processing entry: firebase-web-authn-main/dist/libs/extension/
[debug] [2023-08-25T19:25:33.657Z] [unzip] mkdir: /var/folders/j2/g1020gyn5z56cn_cxf7z8c1w0000gn/T/tmp-21586-pfgwn8f5lo3M/firebase-web-authn-main/dist/libs/extension/
[debug] [2023-08-25T19:25:33.657Z] [unzip] Entry: firebase-web-authn-main/dist/libs/extension/CHANGELOG.md (compressed_size=1150 bytes, uncompressed_size=2976 bytes)
[debug] [2023-08-25T19:25:33.657Z] [unzip] Processing entry: firebase-web-authn-main/dist/libs/extension/CHANGELOG.md
[debug] [2023-08-25T19:25:33.657Z] [unzip] else mkdir: /var/folders/j2/g1020gyn5z56cn_cxf7z8c1w0000gn/T/tmp-21586-pfgwn8f5lo3M/firebase-web-authn-main/dist/libs/extension
[debug] [2023-08-25T19:25:33.657Z] [unzip] deflating: /var/folders/j2/g1020gyn5z56cn_cxf7z8c1w0000gn/T/tmp-21586-pfgwn8f5lo3M/firebase-web-authn-main/dist/libs/extension/CHANGELOG.md

...

[debug] [2023-08-25T19:25:34.080Z] >>> [apiv2][query] GET https://firebaseextensionspublisher.googleapis.com/v1beta/publishers/gavinsawyer/extensions/firebase-web-authn/versions filter=id%3D%2210.1.2%22&showPrereleases=true&pageSize=100&pageToken=
[debug] [2023-08-25T19:25:34.565Z] <<< [apiv2][status] GET https://firebaseextensionspublisher.googleapis.com/v1beta/publishers/gavinsawyer/extensions/firebase-web-authn/versions 200
[debug] [2023-08-25T19:25:34.565Z] <<< [apiv2][body] GET https://firebaseextensionspublisher.googleapis.com/v1beta/publishers/gavinsawyer/extensions/firebase-web-authn/versions {}
[error] 
[error] Error: No entry for version 10.1.2 found in CHANGELOG.md. Please add one so users know what has changed in this version. See https://firebase.google.com/docs/extensions/publishers/user-documentation#writing-changelog for more details.


```

**Comment by pavelgj** [comment]
This was fixed in #6326, please try again.

**Comment by gavinsawyer** [comment]
I was told about that update by customer service and have been able to upload my extension. Meant to close this myself since it's been fixed, thanks!
